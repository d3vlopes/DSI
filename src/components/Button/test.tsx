import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import Button from '.'

describe('<Button />', () => {
  it('should render the primary style and normal size by default', () => {
    renderWithTheme(<Button>Button</Button>)

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '16.1rem',
      'background-color': '#EE4248',
      color: '#FFFFFF',
    })

    expect(button).toHaveStyleRule('background-color', '#de3f45', {
      modifier: ':hover',
    })
  })

  it('should render the large size ', () => {
    renderWithTheme(<Button size="large">Button</Button>)

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '16.6rem',
    })
  })

  it('should render the secodary style ', () => {
    renderWithTheme(<Button styles="secondary">Button</Button>)

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      'background-color': '#FFFFFF',
      color: '#EE4248',
    })

    expect(button).toHaveStyleRule('background-color', '#f7f6f6', {
      modifier: ':hover',
    })
  })

  it('should render the secodary style and large size', () => {
    renderWithTheme(
      <Button size="large" styles="secondary">
        Button
      </Button>,
    )

    const button = screen.getByRole('button', { name: /button/i })

    expect(button).toHaveStyle({
      width: '16.6rem',
      'background-color': '#FFFFFF',
      color: '#EE4248',
    })
  })

  it('should render Button as a link', () => {
    renderWithTheme(
      <Button as="a" href="/link">
        Button
      </Button>,
    )

    expect(screen.getByRole('link', { name: /button/i })).toHaveAttribute(
      'href',
      '/link',
    )
  })
})
