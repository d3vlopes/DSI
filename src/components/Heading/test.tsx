import { render, screen } from 'utils/test-utils'
import { ItemDefaultValues } from 'hooks/use-item'

import Heading from '.'

describe('<Heading />', () => {
  it('should render the Heading with the grey color by default', () => {
    render(<Heading>Children</Heading>)

    expect(screen.getByRole('heading', { name: /children/i })).toHaveStyle({
      color: '#1F2E35',
    })
  })

  it('should render the wavy in red by default ', () => {
    const { container } = render(<Heading wavy>Children</Heading>)

    const wavy = container.querySelector('#wavy')
    expect(wavy).toHaveStyle({
      background: `url('/img/line-red.png') bottom repeat-x`,
    })
  })

  it('should Heading must be rendered with the h1 tag and with wavy', () => {
    const itemProviderProps = {
      ...ItemDefaultValues,
      as: 'h1',
    }

    const { container } = render(<Heading wavy>Children</Heading>, {
      itemProviderProps,
    })

    const h1 = container.querySelector('div > h1')

    expect(h1).toBeInTheDocument()

    const wavy = container.querySelector('#wavy')
    expect(wavy).toHaveStyle({
      background: `url('/img/line-red.png') bottom repeat-x`,
    })
  })

  it('should render the wavy in black color ', () => {
    const { container } = render(
      <Heading wavy wavyColor="littleBlack">
        Children
      </Heading>,
    )

    const wavy = container.querySelector('#wavy')
    expect(wavy).toHaveStyle({
      background: `url('/img/line-black.png') bottom repeat-x`,
    })
  })
})
