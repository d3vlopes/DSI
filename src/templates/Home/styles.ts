import styled, { css } from 'styled-components'
import media from 'styled-media-query'

import * as HeroStyles from 'components/Hero/styles'
import * as TextContentStyles from 'components/TextContent/styles'
import * as ItemStyles from 'components/Item/styles'

export const SectionHero = styled.section`
  ${HeroStyles.Wrapper} {
    ${({ theme }) => css`
      display: flex;
      flex-direction: column-reverse;
      align-items: center;
      padding: 5.4rem ${theme.spacings.small} 4.7rem ${theme.spacings.small};

      ${media.greaterThan('medium')`
        flex-direction: initial;

      `}

      ${media.greaterThan('huge')`
        padding: 5.1rem 0 10rem 0;

      `}
    `}
  }
`

export const HeroContent = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    align-items: center;

    ${media.greaterThan('medium')`
      align-items: flex-start;
    `}

    > h2 {
      font-size: ${theme.font.sizes.large};
      font-weight: ${theme.font.weight.medium};
      max-width: 37.3rem;
      text-align: center;
      margin: ${theme.spacings.xlarge} 0 ${theme.spacings.small} 0;

      ${media.greaterThan('medium')`
        text-align: initial;
        margin: 0 0 ${theme.spacings.medium} 0;
      `}
    }
  `}
`

export const HeroTitle = styled.h2`
  ${({ theme }) => css`
    color: ${theme.colors.grey};

    > span {
      color: ${theme.colors.primary};
    }
  `}
`

export const HeroDescription = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    line-height: 144.7%;
    max-width: 38.3rem;
    text-align: center;
    margin-bottom: ${theme.spacings.small};

    ${media.greaterThan('medium')`
      text-align: initial;
      max-width: 42rem;
      margin-bottom: ${theme.spacings.medium};
    `}

    ${media.greaterThan('huge')`
      max-width: 52rem;
    `}
  `}
`

export const HeroImage = styled.div`
  > img {
    width: 100%;
    max-width: 52.6rem;
  }

  ${media.greaterThan('medium')`
    > img {
      margin-left: 3.2rem;
    }
  `}

  ${media.greaterThan('large')`
    > img {
      margin-left: 5.2rem;
    }
  `}
`
export const SectionAbout = styled.section`
  ${({ theme }) => css`
    ${TextContentStyles.Wrapper} > div {
      display: flex;
      flex-direction: column;
      align-items: center;

      ${media.greaterThan('medium')`
        display: initial;
    `}
    }

    ${TextContentStyles.Wrapper} > div > h1 {
      font-size: ${theme.font.sizes.medium};
      font-weight: ${theme.font.weight.medium};
      line-height: 133.69%;
      color: ${theme.colors.black};
      text-align: center;
      max-width: 36.4rem;
    }

    ${TextContentStyles.Wrapper} > div > p {
      max-width: 35.8rem;

      ${media.greaterThan('medium')`
        max-width: 52rem;
      `}
    }

    ${media.greaterThan('medium')`
      ${TextContentStyles.Wrapper} > div > h1 {
        text-align: initial;
      }
    `}

    ${media.greaterThan('large')`
      ${TextContentStyles.Wrapper} > div > h1 {
        max-width: 45.4rem;
      }
    `}

    ${TextContentStyles.Wrapper} > div > h1 > span {
      color: ${theme.colors.primary};
    }

    ${TextContentStyles.Wrapper} > div > p > span {
      color: ${theme.colors.primary};
      font-weight: ${theme.font.weight.medium};
    }

    ${ItemStyles.Wrapper} {
      display: flex;
      align-items: center;
    }

    ${ItemStyles.Wrapper} > img {
      max-width: 34.9rem;
      height: auto;
    }

    ${media.greaterThan('medium')`
      ${ItemStyles.Wrapper} > img {
        max-width: 45.3rem;
      }
    `}
  `}
`

export const SectionProvides = styled.section`
  ${({ theme }) => css`
    margin: 3.6rem 0 calc(6.4rem - ${theme.spacings.large});

    ${media.greaterThan('large')`
      margin: 9.6rem 0 calc(8.7rem - ${theme.spacings.large});
    `}

    > div > h2 {
      margin-bottom: ${theme.spacings.medium};
      text-align: center;
    }
  `}
`
