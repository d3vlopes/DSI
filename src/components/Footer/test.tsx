import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import Footer from '.'

describe('<Footer />', () => {
  it('should render Footer', () => {
    renderWithTheme(<Footer />)

    expect(screen.getByRole('heading', { name: /Footer/i })).toBeInTheDocument()
  })
})
