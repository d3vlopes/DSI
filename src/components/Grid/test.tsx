import { renderWithTheme } from 'utils/test/helpers'

import { Grid } from '.'

describe('<Grid />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Grid>Children</Grid>)

    expect(container.firstChild).toMatchInlineSnapshot(`
      .c0 {
        display: grid;
        grid-template-columns: repeat(auto-fill,minmax(23rem,1fr));
        grid-gap: 2.4rem;
        margin: 2.4rem 0;
      }

      <div
        class="c0"
      >
        Children
      </div>
    `)
  })
})
