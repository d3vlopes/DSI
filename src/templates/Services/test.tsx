import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import mockHero from 'components/Hero/mock'
import mockItem from 'components/Item/mock'

import Services from '.'

const props = {
  hero: mockHero[1],
  ourService: mockItem[2],
}

jest.mock('next/router', () => ({
  useRouter: jest.fn(() => ({ asPath: '/services' })),
}))

jest.mock('components/Hero', () => {
  return {
    __esModule: true,
    default: function Mock({ children }: { children: React.ReactNode }) {
      return (
        <div id="hero" data-testid="Mock Hero">
          {children}
        </div>
      )
    },
  }
})

jest.mock('components/Item', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock Item"></div>
    },
  }
})

describe('<Services />', () => {
  it('should render Services', () => {
    renderWithTheme(<Services {...props} />)

    expect(screen.getByTestId('Mock Hero')).toBeInTheDocument()
    expect(screen.getByTestId('Mock Item')).toBeInTheDocument()
  })
})
