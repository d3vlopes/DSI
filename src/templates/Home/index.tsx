import Base from 'templates/Base'

import Hero from 'components/Hero'
import Button from 'components/Button'
import Heading from 'components/Heading'
import Item, { ItemProps } from 'components/Item'
import Provides, { ProvideProps } from 'components/Provides'
import { Container } from 'components/Container'

import * as S from './styles'

export type HomeTemplateProps = {
  hero: {
    title: string
    description: string
    src: string
  }
  about: ItemProps
  provides: ProvideProps[]
}

const Home = ({ hero, about, provides }: HomeTemplateProps) => (
  <S.Wrapper>
    <Base>
      <S.SectionHero>
        <Container>
          <Hero>
            <S.HeroContent>
              <S.HeroTitle>
                {hero.title.slice(0, 36)}
                <br />
                <span>{hero.title.slice(36, 58)}</span>
              </S.HeroTitle>
              <S.HeroDescription>{hero.description}</S.HeroDescription>
              <Button>Read More</Button>
            </S.HeroContent>
            <S.HeroImage>
              <img
                src={hero.src}
                alt="A person next to various types of devices"
              />
            </S.HeroImage>
          </Hero>
        </Container>
      </S.SectionHero>

      <S.SectionAbout>
        <Item
          img={about.img}
          alt={about.alt}
          content={about.content}
          color="whiteF6"
        />
      </S.SectionAbout>

      <S.SectionProvides>
        <Heading>Product and Service</Heading>
        <Container>
          <Provides items={provides} />
        </Container>
      </S.SectionProvides>
    </Base>
  </S.Wrapper>
)

export default Home
