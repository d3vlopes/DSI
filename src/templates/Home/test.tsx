import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helpers'

import Home from '.'

import mockHero from 'components/Hero/mock'
import mockAbout from 'components/Item/mock'

const props = {
  hero: mockHero[0],
  about: mockAbout[0],
}

jest.mock('components/Hero', () => {
  return {
    __esModule: true,
    default: function Mock({ children }: { children: React.ReactNode }) {
      return (
        <div id="hero" data-testid="Mock Hero">
          {children}
        </div>
      )
    },
  }
})

jest.mock('components/Item', () => {
  return {
    __esModule: true,
    default: function Mock() {
      return <div data-testid="Mock About"></div>
    },
  }
})

describe('<Home />', () => {
  it('should render Home', () => {
    renderWithTheme(<Home {...props} />)

    expect(screen.getByTestId('Mock Hero')).toBeInTheDocument()
    expect(screen.getByTestId('Mock About')).toBeInTheDocument()
  })
})
