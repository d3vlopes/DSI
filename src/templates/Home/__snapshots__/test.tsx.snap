// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should render Home 1`] = `
.c15 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c5 {
  display: none;
}

.c8 {
  display: none;
}

.c4 {
  width: 100%;
  max-width: 130rem;
  margin-left: auto;
  margin-right: auto;
  padding-left: calc(3.2rem / 2);
  padding-right: calc(3.2rem / 2);
}

.c2 {
  padding: 2.4rem 0;
  z-index: 20;
  box-shadow: 0px 1px 4px rgba(0,0,0,0.25);
}

.c2 .c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c6 {
  color: #1F2E35;
  cursor: pointer;
  width: 2.4rem;
  height: 2.4rem;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c14 > div {
  margin-left: 2.4rem;
}

.c10 .active {
  font-weight: 700;
}

.c12 {
  font-size: 1.8rem;
  color: #1F2E35;
  margin: 0 5rem 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #FFFFFF;
  position: fixed;
  z-index: 20;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  height: 100vh;
  overflow: hidden;
  -webkit-transition: opacity 0.3s ease-in-out;
  transition: opacity 0.3s ease-in-out;
  opacity: 0;
  pointer-events: none;
}

.c16 > svg {
  position: absolute;
  top: 0;
  right: 0;
  margin: 1rem;
  cursor: pointer;
  width: 2.4rem;
  height: 2.4rem;
}

.c16 .c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 3.2rem 0;
}

.c16 .c11 {
  margin-bottom: 1.6rem;
  -webkit-transform: translateY(3rem);
  -ms-transform: translateY(3rem);
  transform: translateY(3rem);
  -webkit-transition: -webkit-transform 0.3s ease-in-out;
  -webkit-transition: transform 0.3s ease-in-out;
  transition: transform 0.3s ease-in-out;
}

.c16 .c13 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c17 {
  font-size: 1.8rem;
  color: #1F2E35;
  margin: 0 5rem 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 400;
}

.c31 {
  color: #263238;
}

.c30 {
  padding-top: 1.6rem;
}

.c32 {
  margin: 0.6rem 0 3.2rem 0;
}

.c32 > h4 {
  font-weight: 500;
  margin-bottom: 1rem;
}

.c32 > div {
  max-width: 31.7rem;
}

.c32 > div > span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 300;
  line-height: 149.19%;
}

.c33 {
  margin-bottom: 3.2rem;
}

.c33 > h3 {
  margin-bottom: 1rem;
  font-size: 2.4rem;
  font-weight: 500;
}

.c33 > div {
  max-width: 26.9rem;
}

.c33 > div > span {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  font-weight: 300;
  line-height: 149.19%;
}

.c34 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
  gap: 1.6rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 3.2rem;
}

.c34 img {
  max-width: min(100%,8.9rem);
}

.c35 {
  padding: 2.4rem 0;
  background-color: #EE4248;
  color: #FFFFFF;
  text-align: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100vh;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c19 {
  margin-top: 3.2rem;
  -webkit-flex: 1 0 auto;
  -ms-flex: 1 0 auto;
  flex: 1 0 auto;
}

.c29 {
  margin-top: 2.9rem;
  padding-top: 3.5rem;
}

.c24 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 1.2rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 2.0rem;
  box-shadow: 1px 2px 4px rgba(0,0,0,0.25);
  -webkit-transition: background-color 0.1s ease-in-out;
  transition: background-color 0.1s ease-in-out;
  width: 16.1rem;
  background-color: #EE4248;
  color: #FFFFFF;
}

.c24:hover {
  background-color: #de3f45;
}

.c28 {
  position: relative;
  font-size: 3.2rem;
  line-height: 133.7%;
  font-weight: 500;
  color: #1F2E35;
}

.c0 .c18 {
  margin-top: 0;
}

.c20 .sc-dmlrTW {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column-reverse;
  -ms-flex-direction: column-reverse;
  flex-direction: column-reverse;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 5.4rem 1.6rem 4.7rem 1.6rem;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c21 > h2 {
  font-size: 3.4rem;
  font-weight: 500;
  max-width: 37.3rem;
  text-align: center;
  margin: 3.2rem 0 1.6rem 0;
}

.c22 {
  color: #1F2E35;
}

.c22 > span {
  color: #EE4248;
}

.c23 {
  font-size: 2.0rem;
  line-height: 144.7%;
  max-width: 38.3rem;
  text-align: center;
  margin-bottom: 1.6rem;
}

.c25 > img {
  width: 100%;
  max-width: 52.6rem;
}

.c26 .sc-fKFyDc > div > h1 {
  font-size: 3.2rem;
  font-weight: 500;
  line-height: 133.69%;
  color: #30444E;
  text-align: center;
}

.c26 .sc-fKFyDc > div > h1 > span {
  color: #EE4248;
}

.c26 .sc-fKFyDc > div > p > span {
  color: #EE4248;
  font-weight: 500;
}

.c26 .sc-bBXqnf {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c26 .sc-bBXqnf > img {
  max-width: 34.9rem;
  height: auto;
}

.c27 {
  margin: 3.6rem 0 calc(6.4rem - 2.9rem);
}

.c27 > h2 {
  margin-bottom: 2.4rem;
  text-align: center;
}

@media (max-width:768px) {
  .c5 {
    display: block;
  }
}

@media (min-width:768px) {
  .c8 {
    display: block;
  }
}

@media (max-width:768px) {
  .c2 .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: justify;
    -webkit-justify-content: space-between;
    -ms-flex-pack: justify;
    justify-content: space-between;
  }
}

@media (max-width:1170px) {
  .c14 {
    display: none;
  }
}

@media (max-width:768px) {
  .c7 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media (max-width:1170px) {
  .c12 {
    margin: 0 2rem 0;
  }
}

@media (min-width:1170px) {
  .c12 {
    margin: 0 3.5rem 0;
  }
}

@media (min-width:768px) {
  .c31 {
    display: grid;
    grid-template-columns: repeat(3,1fr);
    margin-bottom: 1rem;
  }
}

@media (min-width:768px) {
  .c32 {
    margin-top: 1rem;
  }
}

@media (min-width:768px) {
  .c20 .sc-dmlrTW {
    -webkit-flex-direction: initial;
    -ms-flex-direction: initial;
    flex-direction: initial;
  }
}

@media (min-width:1440px) {
  .c20 .sc-dmlrTW {
    padding: 5.1rem 0 10rem 0;
  }
}

@media (min-width:768px) {
  .c21 {
    -webkit-align-items: flex-start;
    -webkit-box-align: flex-start;
    -ms-flex-align: flex-start;
    align-items: flex-start;
  }
}

@media (min-width:768px) {
  .c21 > h2 {
    text-align: initial;
    margin: 0 0 2.4rem 0;
  }
}

@media (min-width:768px) {
  .c23 {
    text-align: initial;
    max-width: 42rem;
    margin-bottom: 2.4rem;
  }
}

@media (min-width:1440px) {
  .c23 {
    max-width: 52rem;
  }
}

@media (min-width:768px) {
  .c25 > img {
    margin-left: 3.2rem;
  }
}

@media (min-width:1170px) {
  .c25 > img {
    margin-left: 5.2rem;
  }
}

@media (min-width:768px) {
  .c26 .sc-fKFyDc > div > h1 {
    text-align: initial;
  }
}

@media (min-width:1170px) {
  .c26 .sc-fKFyDc > div > h1 {
    max-width: 45.4rem;
  }
}

@media (min-width:768px) {
  .c26 .sc-bBXqnf > img {
    max-width: 45.3rem;
  }
}

@media (min-width:1170px) {
  .c27 {
    margin: 9.6rem 0 calc(8.7rem - 2.9rem);
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <menu
      class="c2"
    >
      <div
        class="c3 c4"
      >
        <img
          alt="Letters DSI and words DWIDASA, SAMSARA, INDONESIA"
          class=""
          src="/img/logo.png"
        />
        <div
          class="c5"
        >
          <div
            class="c6"
          >
            <img
              alt="Open Menu"
              class="c7"
              src="/icon/menu-icon.svg"
            />
          </div>
        </div>
        <div
          class="c8"
        >
          <div
            class="c9 c10"
          >
            <a
              class="c11 c12"
              href="/services"
              id="menu-link-services"
              title="Services"
            >
              Services
            </a>
            <a
              class="c11 c12"
              href="/product"
              id="menu-link-product"
              title="Product"
            >
              Product
            </a>
            <a
              class="c11 c12"
              href="/technology"
              id="menu-link-technology"
              title="Technology"
            >
              Technology
            </a>
            <a
              class="c11 c12"
              href="/about"
              id="menu-link-about"
              title="About"
            >
              About
            </a>
            <a
              class="c11 c12"
              href="/client"
              id="menu-link-client"
              title="Client"
            >
              Client
            </a>
            <a
              class="c11 c12"
              href="/partner"
              id="menu-link-partner"
              title="Partner"
            >
              Partner
            </a>
          </div>
        </div>
        <div
          class="c13 c14"
          id="menu-group"
        >
          <div
            class="c6"
            id="menu-group-icon"
          >
            <svg
              aria-hidden="true"
              aria-label="Go to home"
              class="c15"
              fill="none"
              focusable="false"
              id="menu-home-icon"
              stroke="currentColor"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <div
            class="c6"
            id="menu-group-icon"
          >
            <svg
              aria-hidden="true"
              aria-label="Send message"
              class="c15"
              fill="currentColor"
              focusable="false"
              id="menu-email-icon"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                data-name="Layer 2"
              >
                <path
                  d="M19 4H5a3 3 0 00-3 3v10a3 3 0 003 3h14a3 3 0 003-3V7a3 3 0 00-3-3zm-.67 2L12 10.75 5.67 6zM19 18H5a1 1 0 01-1-1V7.25l7.4 5.55a1 1 0 00.6.2 1 1 0 00.6-.2L20 7.25V17a1 1 0 01-1 1z"
                  data-name="email"
                />
              </g>
            </svg>
          </div>
          <div
            class="c6"
            id="menu-group-icon"
          >
            <svg
              aria-hidden="true"
              aria-label="Media"
              class="c15"
              fill="currentColor"
              focusable="false"
              id="menu-shuffle-icon"
              viewBox="0 0 24 24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
              />
            </svg>
          </div>
        </div>
      </div>
      <nav
        aria-hidden="true"
        class="c16"
      >
        <svg
          aria-hidden="true"
          aria-label="Close Menu"
          class="c15"
          fill="currentColor"
          focusable="false"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
          />
          <path
            d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"
          />
        </svg>
        <div>
          <div
            class="c9 c10"
          >
            <a
              class="c17"
              href="/services"
              id="menufull-link-services"
              title="Services"
            >
              Services
            </a>
            <a
              class="c17"
              href="/product"
              id="menufull-link-product"
              title="Product"
            >
              Product
            </a>
            <a
              class="c17"
              href="/technology"
              id="menufull-link-technology"
              title="Technology"
            >
              Technology
            </a>
            <a
              class="c17"
              href="/about"
              id="menufull-link-about"
              title="About"
            >
              About
            </a>
            <a
              class="c17"
              href="/client"
              id="menufull-link-client"
              title="Client"
            >
              Client
            </a>
            <a
              class="c17"
              href="/partner"
              id="menufull-link-partner"
              title="Partner"
            >
              Partner
            </a>
          </div>
          <div
            class="c13 c14"
            id="mobile-menu-group"
          >
            <div
              class="c6"
            >
              <svg
                aria-hidden="true"
                aria-label="Go to home"
                class="c15"
                fill="none"
                focusable="false"
                stroke="currentColor"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="c6"
            >
              <svg
                aria-hidden="true"
                aria-label="Send message"
                class="c15"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  data-name="Layer 2"
                >
                  <path
                    d="M19 4H5a3 3 0 00-3 3v10a3 3 0 003 3h14a3 3 0 003-3V7a3 3 0 00-3-3zm-.67 2L12 10.75 5.67 6zM19 18H5a1 1 0 01-1-1V7.25l7.4 5.55a1 1 0 00.6.2 1 1 0 00.6-.2L20 7.25V17a1 1 0 01-1 1z"
                    data-name="email"
                  />
                </g>
              </svg>
            </div>
            <div
              class="c6"
            >
              <svg
                aria-hidden="true"
                aria-label="Media"
                class="c15"
                fill="currentColor"
                focusable="false"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M10.59 9.17L5.41 4 4 5.41l5.17 5.17 1.42-1.41zM14.5 4l2.04 2.04L4 18.59 5.41 20 17.96 7.46 20 9.5V4h-5.5zm.33 9.41l-1.41 1.41 3.13 3.13L14.5 20H20v-5.5l-2.04 2.04-3.13-3.13z"
                />
              </svg>
            </div>
          </div>
        </div>
      </nav>
    </menu>
    <div
      class="c18 c19"
    >
      <section
        class="c20"
      >
        <div
          class="c3 c4"
        >
          <div
            data-testid="Mock Hero"
            id="hero"
          >
            <div
              class="c21"
            >
              <h2
                class="c22"
              >
                Making the most of the ever-growing 
                <br />
                <span>
                  Information Technology
                </span>
              </h2>
              <p
                class="c23"
              >
                Managed by a team of professional experts with extensive experience and impressive track records
              </p>
              <button
                class="c24"
              >
                <span>
                  Read More
                </span>
              </button>
            </div>
            <div
              class="c25"
            >
              <img
                alt="A person next to various types of devices"
                src="img/hero/home.png"
              />
            </div>
          </div>
        </div>
      </section>
      <section
        class="c26"
      >
        <div
          data-testid="Mock About"
        />
      </section>
      <section
        class="c27"
      >
        <h2
          class="c28"
          color="grey"
        >
          Product and Service
        </h2>
        <div
          class="c3 c4"
        >
          <div
            data-testid="Mock Provides"
          />
        </div>
      </section>
    </div>
    <section
      class="c29"
    >
      <footer
        class=""
      >
        <div
          class="c3 c4"
        >
          <img
            alt="Letters DSI and words DWIDASA, SAMSARA, INDONESIA"
            class="c30"
            src="/img/logo.png"
          />
          <div
            class="c31"
          >
            <div
              class="c32"
            >
              <h4>
                PT Dwidasa Samsara Indonesia
              </h4>
              <div>
                <span>
                  Ruko Jalur Sutera 29A No. 53
                </span>
                <span>
                  Alam Sutera Serpong, Tangerang 15323
                </span>
              </div>
            </div>
            <div
              class="c33"
            >
              <h3>
                Contact
              </h3>
              <div>
                <span>
                  Phone : +62.21.5314.1135
                </span>
                <span>
                  Fax : +62.21.5314.1135
                </span>
                <span>
                  Email : community@dwidasa.com
                </span>
              </div>
            </div>
            <div
              class="c34"
            >
              <img
                alt="icon of Microsoft Silverlight"
                id="platform-0"
                src="/img/platform/microsoft.png"
              />
              <img
                alt="icon of Microsoft .Net Framework"
                id="platform-1"
                src="/img/platform/net.png"
              />
              <img
                alt="icon of Java"
                id="platform-2"
                src="/img/platform/java.png"
              />
              <img
                alt="icon of IOS"
                id="platform-3"
                src="/img/platform/ios.png"
              />
              <img
                alt="icon of Android"
                id="platform-4"
                src="/img/platform/android.png"
              />
              <img
                alt="icon of Black Berry"
                id="platform-5"
                src="/img/platform/black-berry.png"
              />
            </div>
          </div>
        </div>
        <div
          class="c35"
        >
          Copyright © 2021 - Dwidasa Samsara Indonesia
        </div>
      </footer>
    </section>
  </div>
</div>
`
